### PR TITLE
[CURATOR-314] Listenable support for GroupMember recipe

### PR DESCRIPTION
**Pull request by shivramk**
I've added support for listenable events to the GroupMember recipe. Its built on the event support provided by the underlying PathChildenCache object being used by GroupMember

### CODE REVIEW

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupData.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/**
```
Comment (Randgalt) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupData.java]: Can't ChildData be re-used here? The only difference is the stat object which doesn't hurt for GroupMember.

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupMember.java [hunk 2]
```diff
@@ -75,6 +119,7 @@ public void start()
         try
         {
             cache.start();
+            cache.getListenable().addListener(childrenCacheListener);
```
Comment (Randgalt) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupMember.java]: Why not add a proxy listener getListenable()? I need to think about this. Does this single listener correctly proxy all behaviors from getListenable()?

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupMemberEvent.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.curator.framework.recipes.nodes;
+
+/**
+ * POJO that abstracts a change to a group membership
+ */
+public class GroupMemberEvent
```
Comment (Randgalt) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/GroupMemberEvent.java]: PathChildrenCacheEvent or TreeCacheEvent can be used, right? I'm leery expanded the number of event objects.

**Comment by tisonkun** [comment]
Closed as stale. There're many changes since this patch was made in the first place. Feel free to resubmit it if it's still relevant.

**Pull request closed without merging**